upgrade/1.5.1/README.md

```diff
@@ -0,0 +1,316 @@
+monitoring daemons will be redeployed so that they use these images in Nexus. Once this is complete, all Ceph daemons
+should be using images in Nexus and not images hosted in the local Docker registry on storage nodes. The third step
+stops the local Docker registry on all storage nodes.
+
+1. (`ncn-m001#`) Run Ceph upgrade to `v16.2.13`.
+
+   ```bash
+   /usr/share/doc/csm/upgrade/scripts/ceph/ceph-upgrade-tool.py --version "v16.2.13"
+   ```
+
+1. (`ncn-m001#`) Redeploy Ceph monitoring daemons so they are using images in Nexus.
+
+   ```bash
+   scp /usr/share/doc/csm/scripts/operations/ceph/redeploy_monitoring_stack_to_nexus.sh ncn-s001:/srv/cray/scripts/common/redeploy_monitoring_stack_to_nexus.sh
+   ssh ncn-s001 "/srv/cray/scripts/common/redeploy_monitoring_stack_to_nexus.sh"
+   ```
+
+1. (`ncn-m001#`) Stop the local Docker registries on all storage nodes.
+
+   ```bash
+   scp /usr/share/doc/csm/scripts/operations/ceph/disable_local_registry.sh ncn-s001:/srv/cray/scripts/common/disable_local_registry.sh
+   ssh ncn-s001 "/srv/cray/scripts/common/disable_local_registry.sh"
+   ```
+
+### Enable `smartmon` metrics on storage NCNs
+
+This step will install the `smart-mon` rpm on storage nodes, and reconfigure the `node-exporter` to provide `smartmon`
+metrics.
+
+1. (`ncn-m001#`) Execute the following script.
+
+   ```bash
+   /usr/share/doc/csm/scripts/operations/ceph/enable-smart-mon-storage-nodes.sh
+   ```
+
+### Update management node CFS configuration
+
+This step updates the CFS configuration which is set as the desired configuration for the management
+nodes (NCNs). It ensures that the CFS configuration layers reference the correct commit hash for the
+version of CSM being installed. It then waits for the components to reach a configured state in CFS.
+
+1. (`ncn-m001#`)
+
+   ```bash
+   cd "$CSM_DISTDIR"
+   ./update-mgmt-ncn-cfs-config.sh --base-query role=management \
+      --save --create-backups --clear-error
+   ```
+
+   The output will look similar to the truncated output shown below.
+
+   ```text
+   INFO: Querying CFS configurations for the following NCNs: x3000c0s5b0n0, ...
+   INFO: Found configuration "management-csm-1.4.0" for component x3000c0s5b0n0
+   ...
+   INFO: Updating existing layer with repo path /vcs/cray/csm-config-management.git and playbook site.yml
+   INFO: Property "commit" of layer with repo path /vcs/cray/csm-config-management.git and playbook site.yml updated ...
+   INFO: Property "name" of layer with repo path /vcs/cray/csm-config-management.git and playbook site.yml updated ...
+   INFO: No layer with repo path /vcs/cray/csm-config-management.git and playbook ncn-initrd.yml found.
+   INFO: Adding a layer with repo path /vcs/cray/csm-config-management.git and playbook ncn-initrd.yml to the end.
+   INFO: Successfully saved CFS configuration "management-csm-1.4.0-backup-20230918T205149"
+   INFO: Successfully saved CFS configuration "management-csm-1.4.0"
```

Comment:
Ideally we should just get actual output from an upgrade, but failing that, this makes the changes that I can anticipate:

```suggestion
   INFO: Found configuration "management-csm-1.5.0" for component x3000c0s5b0n0
   ...
   INFO: Updating existing layer with repo path /vcs/cray/csm-config-management.git and playbook ncn_nodes.yml
   INFO: Property "commit" of layer with repo path /vcs/cray/csm-config-management.git and playbook ncn_nodes.yml updated ...
   INFO: Property "name" of layer with repo path /vcs/cray/csm-config-management.git and playbook ncn_nodes.yml updated ...
   INFO: No layer with repo path /vcs/cray/csm-config-management.git and playbook ncn-initrd.yml found.
   INFO: Adding a layer with repo path /vcs/cray/csm-config-management.git and playbook ncn-initrd.yml to the end.
   INFO: Successfully saved CFS configuration "management-csm-1.5.0-backup-20240410T205149"
   INFO: Successfully saved CFS configuration "management-csm-1.5.0"
```